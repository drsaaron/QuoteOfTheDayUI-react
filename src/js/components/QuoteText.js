/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import React, {Component} from 'react';
import sanitizeHtml from 'sanitize-html-react';

export default class QuoteText extends Component {

    render() {
        var quote = this.props.quote;
        var quoteText = (quote.text) ? sanitizeHtml(quote.text.replace(new RegExp('\r?\n', 'g'), '<br />')) : "<span></span>";

        return (
                <div dangerouslySetInnerHTML={{__html: quoteText}} className="QuoteText"></div>
                );
    }
}
