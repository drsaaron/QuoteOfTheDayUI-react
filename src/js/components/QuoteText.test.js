import React from 'react';
import renderer from 'react-test-renderer';

const goHome = (navigate) => {};

import QuoteText from './QuoteText';

describe('header', () => {
    test('basic', () => {
	const quote = { text: 'I am a quote\nHear me roar' };
	const component = renderer.create(<QuoteText quote={quote} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
