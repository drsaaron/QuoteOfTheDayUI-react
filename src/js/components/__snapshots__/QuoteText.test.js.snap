// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header basic 1`] = `
<div
  className="QuoteText"
  dangerouslySetInnerHTML={
    Object {
      "__html": "I am a quote<br />Hear me roar",
    }
  }
/>
`;
